### PR TITLE
🌪🌈 Complete Farmhouse: replace Arrow with Icon

### PR DESCRIPTION
**Pull request by whimsicallyson**
./src/components/collection/collection-item.js:9/73
./src/components/collection/collection-item-small.js:9/73
./src/components/collection/container.js:9/73
./src/components/collections-list/more-collections.js:9/114
./src/components/containers/results-list.js:9/79
./src/components/new-stuff/new-stuff-article.js:9/125
./src/components/questions/index.js:9/125
./src/components/recent-projects/index.js:9/73
./src/components/related-projects/index.js:9/168
./src/components/search-form/autocomplete.js:9/126
./src/presenters/pages/home-v2/banner.js:9/73
./src/presenters/pages/home-v2/index.js:9/178

## Links
* https://complete-farmhouse.glitch.me/
* https://app.zenhub.com/workspaces/community-5d07d1359dee6059a9688176/issues/fogcreek/glitch-community-work/392

## Changes:
* the Arrow component is really just a (not so) special case of Icon, so just replace it everywhere

## How To Test:
* Click around, look for arrows (there are a bunch on the logged-out homepage, also handraise has one)
* Check spacing, how they look on different browsers

## Things left to do before deploying:
- Marvel at how `complete-farmhouse` had the checkin of tornado + rainbow? are we off to see the wizard?

### CODE REVIEW

**Comment by GregWeil** [comment]
image

It's pulling a `width: 100%` style that's applied to all svgs in the banner

**Comment by GregWeil** [comment]
image
The arrow is positioned a bit high in section titles

**Comment by GregWeil** [comment]
image
Also when you hover over the text it underlines the space between the text and the arrow

**Comment by GregWeil** [comment]
It's at the bottom of the collection page

**Comment by whimsicallyson** [comment]
Gotcha. That style of link doesn't get underlined on a project page or the homepage, so I just removed the underlining entirely

File: src/components/collection/collection-item.styl [hunk 1]
```diff
@@ -184,4 +184,7 @@ a:hover .smallProjectCount
   display: inline-block
   width: 50%
   vertical-align: top
-  margin: 20px
\ No newline at end of file
+  margin: 20px
+  
+svg.arrow
+  vertical-align: text-top
```
Comment (GregWeil) [inline src/components/collection/collection-item.styl]: `text-top` puts it in the same position as it was before

Comment (whimsicallyson) [reply src/components/collection/collection-item.styl]: its actually a bit lower (`top` vs `text-top`) but `middle` is closer to prod, so I updated.

File: src/components/collection/collection-item.styl [hunk 2]
```diff
@@ -184,4 +184,7 @@ a:hover .smallProjectCount
   display: inline-block
   width: 50%
   vertical-align: top
-  margin: 20px
\ No newline at end of file
+  margin: 20px
+  
+svg.arrow
+  vertical-align: middle
```
Comment (ttseng) [inline src/components/collection/collection-item.styl]: I think this should be `vertical-align: text-top` or else the arrow doesn't quite look centered vertically next to the View All text: (text-top on the left, middle on the right)

image

**Comment by ttseng** [comment]
The badges in buttons are missing their left/right margins:
image

File: src/components/related-projects/styles.styl [hunk 3]
```diff
@@ -4,3 +4,6 @@
   list-style-type: none
 .projectsWrap
   margin: -1rem 0
+  
+svg.arrow
+  vertical-align: middle
```
Comment (ttseng) [inline src/components/related-projects/styles.styl]: this should also be `vertical-align: text-top`

(with vertical-middle) https://complete-farmhouse.glitch.me/~palm-organ
image

File: src/components/containers/results-list.styl [hunk 4]
```diff
@@ -54,4 +54,7 @@
   position: absolute
   top: 0
   right: 0
-  padding: 12px 6px
\ No newline at end of file
+  padding: 12px 6px
+  
+svg.arrow
+  vertical-align: middle
```
Comment (ttseng) [inline src/components/containers/results-list.styl]: `vertical-align: text-top`

File: src/components/search-form/autocomplete.styl [hunk 5]
```diff
@@ -68,3 +68,6 @@
 .memberContainer
   flex: 0 0 100px
   padding-left: 4px
+
+svg.arrow
+  vertical-align: middle
```
Comment (ttseng) [inline src/components/search-form/autocomplete.styl]: `vertical-align: text-top`

**Comment by whimsicallyson** [comment]
@ttseng badges were updated in https://github.com/FogCreek/Glitch-Community/pull/903, so should be fine once everything's merged 👍 thanks for finding where we want to use `text-top` vs `middle`!

**Comment by ttseng** [comment]
In the remix, it looks like there's still flashing of the arrow component on the homepage:

loading

But it should be fixed now, so if you pull the latest changes, it should be fine!  Do you mind doing that and then I can re-review?

**Comment by whimsicallyson** [comment]
@ttseng updated, looks like that should have fixed it :)

**Pull request merged**